File: src/dotnet/Orders.Api/Services/CarrierService.cs
using Orders.Application.Interfaces;

namespace Orders.Api.Services;

// Service to simulate having a real carrier service to check shipped / delivered status of orders
public class CarrierService : ICarrierService
{
    private static readonly Random Random = new Random();

    public bool IsOrderShipped(Guid orderId)
    {
        return Random.Next(0, orderId.ToString().Length) < orderId.ToString().Length / 4;
    }

    public bool IsOrderDelivered(Guid orderId)
    {
        return Random.Next(0, orderId.ToString().Length) < orderId.ToString().Length / 4;
    }
}

File: src/laravel/app/Services/CarrierService.php
<?php

namespace App\Services;

use App\Services\Interfaces\ICarrierService;

class CarrierService implements ICarrierService
{
    public function isOrderShipped(string $orderId): bool
    {
        return random_int(0, strlen($orderId)) < strlen($orderId) / 4;
    }

    public function isOrderDelivered(string $orderId): bool
    {
        return random_int(0, strlen($orderId)) < strlen($orderId) / 4;
    }
}
